karoo_gp/base_class.py

```diff
@@ -1787,1535 +2043,1588 @@ class labels provided in the data .csv. Outputs an array of tuples
+# used by: Population
+def fx_evolve_child_insert(tree, node, c_buffer, error):
 
-        ### THIS METHOD MAY NOT BE REQUIRED AS SORTING 'branch' SEEMS TO HAVE FIXED 'parent_id' ###
+    '''
+    Insert child node into the copy of a parent Tree.
 
-        # tested 2015 06/05
-        for node in range(1, len(tree[3])):
+    Called by: fx_evolve_branch_insert
 
-            if tree[9][node] != '':
-                child = int(tree[9][node])
-                tree[7][child] = node
+    Arguments required: tree, node, c_buffer
+    '''
 
-            if tree[10][node] != '':
-                child = int(tree[10][node])
-                tree[7][child] = node
+    if int(tree.root[8][node]) == 0:  # if arity = 0
+        error(f'\n\t\033[31m ERROR! In fx_evolve_child_insert: node '
+              f'{node} has arity 0\033[0;0m')
+        # # consider special instructions for this (pause) - 2019 06/08
+        # self.fx_karoo_pause()
+
+    elif int(tree.root[8][node]) == 1:  # if arity = 1
+        # insert node for 'node_c1'
+        tree.root = np.insert(tree.root, c_buffer, '', axis=1)
+        tree.root[3][c_buffer] = c_buffer  # node ID
+        tree.root[4][c_buffer] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer] = int(tree.root[3][node])  # parent ID
+
+    elif int(tree.root[8][node]) == 2:  # if arity = 2
+        # insert node for 'node_c1'
+        tree.root = np.insert(tree.root, c_buffer, '', axis=1)
+        tree.root[3][c_buffer] = c_buffer  # node ID
+        tree.root[4][c_buffer] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer] = int(tree.root[3][node])  # parent ID
+
+        # insert node for 'node_c2'
+        tree.root = np.insert(tree.root, c_buffer + 1, '', axis=1)
+        tree.root[3][c_buffer + 1] = c_buffer + 1  # node ID
+        tree.root[4][c_buffer + 1] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer + 1] = int(tree.root[3][node])  # parent ID
+
+    elif int(tree.root[8][node]) == 3:  # if arity = 3
+        # insert node for 'node_c1'
+        tree.root = np.insert(tree.root, c_buffer, '', axis=1)
+        tree.root[3][c_buffer] = c_buffer  # node ID
+        tree.root[4][c_buffer] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer] = int(tree.root[3][node])  # parent ID
+
+        # insert node for 'node_c2'
+        tree.root = np.insert(tree.root, c_buffer + 1, '', axis=1)
+        tree.root[3][c_buffer + 1] = c_buffer + 1  # node ID
+        tree.root[4][c_buffer + 1] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer + 1] = int(tree.root[3][node])  # parent ID
+
+        # insert node for 'node_c3'
+        tree.root = np.insert(tree.root, c_buffer + 2, '', axis=1)
+        tree.root[3][c_buffer + 2] = c_buffer + 2  # node ID
+        tree.root[4][c_buffer + 2] = int(tree.root[4][node]) + 1  # node_depth
+        tree.root[7][c_buffer + 2] = int(tree.root[3][node])  # parent ID
```

Comment:
Can't these 3 blocks replaced with a `for n in range(3):` and then use e.g. `tree.root[3][c_buffer + n] = c_buffer + n`?